torch/csrc/jit/script/compiler.cpp

```diff
@@ -1337,17 +1337,15 @@ struct to_ir {
         Compound::create(TK_LIST, loc, std::move(inputs));
     const auto input_values = getValues(applyInputs->trees());
     Value* tensor = input_values[0];
-    const auto& begin = at::Scalar(input_values[1]->node()->t(attr::value)).toInt();
-    const auto& end = at::Scalar(input_values[2]->node()->t(attr::value)).toInt();
+    Value* begin = input_values[1];
+    Value* end = input_values[2];
+    Value* zero_dim = emitConst(Const::create(loc, std::to_string(0)));
+    Value* one_step = emitConst(Const::create(loc, std::to_string(1)));
     return emitNode(
                Symbol::aten("slice"),
                loc,
-               {tensor},
-               1)
-               ->i_(attr::dim, 0)
-               ->i_(attr::step, 1)
-               ->i_(attr::start, begin)
-               ->i_(attr::end, end)->output();
+               {tensor, zero_dim, begin, end, one_step},
+               1)->output();
```

Comment:
I'd be nice to call `liftConstantAttributes` so that we get the attribute version of the function in the case it is constant. You may have to lookup the FunctionSchema for "slice" to call it.